### ISSUE TITLE
maxlength attribute in text_field_tag adds unwanted size attribute

### ISSUE THREAD

**Issue by rowend** [rails/rails#34291]
### Description [ActionView/Helpers/text_field_tag][html]

The *maxlength* option used in the text_field helper adds an unwanted *size* attribute.

### Steps to reproduce

1) Generate a form using the form_with helper and use the **maxlength** attribute in the **text_field** helper:

```
<%= form_with model: any_model do |f| %>
  <div>
    <%= f.label :name %>
    <%= f.text_field :name, maxlength: 100 %>
  </div>
  <div><%= f.submit %></div>
<% end %>
```

### Expected behavior

The generated html for the input only must add the maxlength attribute.

```
<input maxlength="100"  type="text" value="" name="any_model[name]" id="any_model_name">
```

### Actual behavior

Adds the maxlength attribute and the unwanted **size** attribute.

```
<input maxlength="100"  size="100" type="text" value="" name="any_model[name]" id="any_model_name">
```

### System configuration

**OS version**: macos High Sierra 10.13.5

**Rails version**: 5.2.1

**Ruby version**: ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin17]

### Notes

* I am not sure if this is a bug but I searched in the documentation but I did not find anything related to this behavior.
* If you add the *size* attribute it does not add the *maxlength* attribute.

Thank you for the support and all the work you do.

**Comment by amree** [comment]
If I may add, this is only going to affect `text_field` tag helper in the form block, but not `text_field_tag`.

It's 'documented' in the [test](https://github.com/rails/rails/blob/5-2-1/actionview/test/template/form_helper_test.rb#L512-L522) with a test named as `test_text_field_assuming_size`. So, I'm guessing it's trying to assume the size based on the `maxlength`.

For now, you can remove it by explicitly setting the `size` to `nil`. Maybe we can make it consistent by removing the assumption.

**Comment by GeniusStorm** [comment]
I believe that this should be documented as an example in the rails api so that people who run into this dont endup thinking this as a bug ( atleast until someone modifies the behaviour to assume size to be same as maxlength is size is not provided). I can create a quick PR for the documentation if someone from the rails team approves for the documentation.

**Comment by y-yagi** [comment]
Fixed by #34313
